sdk/packages/selector-polkadot/src/adapter.ts

```diff
@@ -0,0 +1,490 @@
+            return
+          }
+
+          this._connecting = true
+          if (this._app.hasBeenRestored() && this._app.accounts.activeAccounts.length > 0) {
+            this.eagerConnectDeeplink()
+            this._connected = true
+            this._connecting = false
+            this._appSessionActive = true
+            resolve()
+            return
+          }
+
+          const recentName = getRecentStandardWalletForNetwork(POLKADOT_NETWORK)
+          if (
+            this._eagerConnectForStandardWallets &&
+            recentName !== null &&
+            isStandardConnectedForNetwork(POLKADOT_NETWORK)
+          ) {
+            await this.connectToStandardWallet(recentName, resolve)
+
+            if (this._connected) {
+              return
+            }
+          }
+
+          this._app.on('userConnected', (e) => {
+            if (this._chosenMobileWalletName) {
+              persistRecentStandardWalletForNetwork(this._chosenMobileWalletName, POLKADOT_NETWORK)
+            } else {
+              clearRecentStandardWalletForNetwork(POLKADOT_NETWORK)
+            }
+            this._connected = true
+            this._connecting = false
+            this._appSessionActive = true
+            this._modal?.closeModal()
+            resolve()
+          })
+
+          if (this._modal) {
+            this._modal._onClose = () => {
+              if (this._connecting) {
+                this._connecting = false
+                const error = new WalletWindowClosedError()
+                reject(error)
+              }
+            }
+            this._modal.openModal(this._app.sessionId, (walletName) => {
+              if (
+                isMobileBrowser() &&
+                !this._walletsList.find((w) => w.name === walletName)?.standardWallet
+              ) {
+                this.connectToMobileWallet(walletName)
+              } else {
+                this.connectToStandardWallet(walletName, resolve)
+              }
+            })
+          }
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          this._connecting = false
+          reject(error)
+        }
+      }
+
+      innerConnect()
+    })
+
+  disconnect = async () => {
+    if (this.connected) {
+      if (this._appSessionActive) {
+        clearSessionIdForNetwork(POLKADOT_NETWORK)
+        this._appSessionActive = false
+        this._loading = true
+        AppPolkadot.build(this._appInitData)
+          .then(
+            (app) => {
+              this._app = app
+            },
+            (err) => {
+              console.log(err)
+            }
+          )
+          .finally(() => {
+            this._loading = false
+          })
+      }
+      if (this._innerStandardAdapter) {
+        this._innerStandardAdapter = undefined
+        persistStandardDisconnectForNetwork(POLKADOT_NETWORK)
+      }
+      if (this._modal) {
+        this._modal.walletsList = this._walletsList
```

Comment:
before this, wallets list should be updated to have info about new recent wallet